src/Nest/Mapping/DynamicTemplate/SingleMapping.cs

```diff
@@ -247,6 +250,30 @@ public IProperty Object<TChild>(Func<ObjectTypeDescriptor<T, TChild>, IObjectPro
 		) =>
 			selector.InvokeOrDefault(new DatePropertyDescriptor<T>().Name(field));
 
+		public IProperty ScalarNanos(Expression<Func<T, DateTime>> field, Func<DateNanosPropertyDescriptor<T>, IDateNanosProperty> selector = null) =>
```

Comment:
@russcam - not sure if you saw this, but I had to rename the method to avoid conflicts with the implementation for `Date`. As `date_nanos` and `date` types both use `DateTime` objects in C# there are places where the two property types conflict when method resolution is attempted by the compiler.

Reply:
saw it 🙂 unavoidable here, I think